test/helpers/test.go

```diff
@@ -171,6 +177,55 @@ func (t *TestOptions) GetGitProvider() (gits.GitProvider, error) {
 	return gitProvider, nil
 }
 
+// GetApproverGitProvider returns a git provider that uses credentials for the approver user defined in environment variables
+// We don't use standard user auth here because the user/token isn't defined during boot, so we'll create the credentials
+// for the user on the fly.
+func (t *TestOptions) GetApproverGitProvider() (gits.GitProvider, error) {
+	factory := cmd.NewFactory()
+	_, ns, err := factory.CreateKubeClient()
+	if err != nil {
+		return nil, err
+	}
+
+	authConfigService, err := factory.CreateAuthConfigService("gitAuth.yaml", ns, kube.ValueKindGit, "")
```

Comment:
There is actually a `CreateGitAuthConfigService` as well in the factory which might be more explicit here.

Reply:
So what is this actually doing? I am confused about what's going on here. Is this supposed to create the _gitAuth.yaml_ or load from it?

Reply:
I never know exactly. Note that this is copied over from the other function - but yeah, I think `CreateGitAuthConfigService` would probably do the same thing in a less confusing way.